packages/key4hep-stack/package.py

```diff
@@ -4,84 +4,238 @@
+    depends_on('lich@master', when='@nightly')
+
+    depends_on('lccd')
+    depends_on('lccd@master', when='@nightly')
+
+    depends_on('lcio')
+    depends_on('lcio@master', when="@nightly")
+
+    depends_on('lcgeo')
+    depends_on('lcgeo@master', when="@nightly")
+
+    depends_on('marlin')
+    depends_on('marlin@master', when="@nightly")
+
+    depends_on('marlinutil')
+    depends_on('marlinutil@master', when="@nightly")
+
+    depends_on('marlinpandora')
+    depends_on('marlinpandora@master', when="@nightly")
+
+    depends_on("marlindd4hep")
+    depends_on('marlindd4hep@master', when='@nightly')
+
+    depends_on('marlinreco')
+    depends_on('marlinreco@master', when='@nightly')
+
+    depends_on('marlinfastjet')
+    depends_on('marlinfastjet@master', when='@nightly')
+
+    depends_on('marlinkinfit')
+    depends_on('marlinkinfit@master', when='@nightly')
+
+    depends_on('marlintrkprocessors')
+    depends_on('marlintrkprocessors@master', when='@nightly')
+
+    depends_on('marlintrk')
+    depends_on('marlintrk@master', when='@nightly')
+
+    depends_on('overlay')
+    depends_on('overlay@master', when='@nightly')
+
+    depends_on('pandoraanalysis')
+    depends_on('pandoraanalysis@master', when='@nightly')
+
+    depends_on('pandorapfa')
+    depends_on('pandorapfa@master', when="@nightly")
+
+    depends_on('physsim')
+    depends_on('physsim@master', when="@nightly")
+
+    depends_on("raida")
+    depends_on('raida@master', when='@nightly')
+
+    depends_on('sio')
+    depends_on('sio@master', when='@nightly')
+
+
+    ############################### fccsw #################
+    #######################################################
+    depends_on("fccsw")
+    depends_on("fccsw@master", when='@nightly')
+
+    depends_on("fcc-edm")
+    depends_on("fcc-edm@master", when="@nightly")
+
+    depends_on("dual-readout")
+    depends_on("dual-readout@master", when="@nightly")
+
+    ##################### developer tools #################
+    #######################################################
+    depends_on("cmake", when="+devtools")
+    depends_on("gdb", "when=+devtools")
+    depends_on("emacs+X toolkit=athena", when="+devtools")
+    depends_on("ninja", when="+devtools")
+    depends_on("py-ipython", when="+devtools")
+
+    ##################### environment boostrap ############
+    #######################################################
+    depends_on("environment-modules", when="+bootstrap")
+
+    ##################### concretizer hints ###############
+    #######################################################
+    depends_on('boost +python', when="+concretizer-hints")
+    # avoid building doxygen
+    depends_on('range-v3 ~doc', when="+concretizer-hints")
+    # avoid building mariadb
+    depends_on('marlin ~lccd', when="+concretizer-hints")
+    # avoid building llvm
+    depends_on('mesa ~llvm+glx+opengl', when="+concretizer-hints")
+    depends_on('dd4hep +geant4', when="+concretizer-hints")
```

Comment:
```suggestion
    depends_on('dd4hep +geant4 +lcio', when="+concretizer-hints")
```
Or how is the lcio variant injected at the moment?

Reply:
good catch!